### DISCUSSION TITLE
Suspend and time management

### DISCUSSION THREAD

**Discussion by gabuzom** [351ELEC/351ELEC#926] [Ideas]
Sorry i tried to discuss this on IRC, but there was nobody there and the channel wasn't bridged to Discord(tm) that I do not consent to the end-user terms of...

Feel free to erase this if it has been already discussed or is considered not relevant.

### Description and steps to reproduce

AmberElec draws quite some power when on suspend, which is OK. But often power is drawn to full exhaustion of the  battery. It happens to me regularly. 

When this happens though, the system reboots with a system time set to 01/01/1970 (good'ol epoch) and a bunch of things start dysfunctioning: list of recently played games, "last played on" and a few others... As my device is used mostly offline, it can stay in that state for a while...

### Expected behavior

One can think of a few ways to mitigate this, that would depend on the power-management capcities of the device:
- either to wake-up when power is -say- 10% to shutdown the system properly. Next time it boots time is likely to have been saved
- or (something maybe more device-agnostic), to regularly save the time somewhere, and as a default load this time upon next boot. It wil be imperfect until next NTP update, but at least all the chronologically updated lists and items would stay in order
- and/or a manual way to set the time from a menu;

### Device (please complete the following information):
20220629_2016
RG351M

### Additional context
AmberElec rocks! <3 Thanks for the amazing work!

**Comment by gabuzom** [comment]
Basically what is happening is that offline device (like mine was for 4 weeks i traveled without a wifi dongle) will inevitably lose the capacity to display "last played" games, tell you when you played a game last, and a few other useful features! 

(as battery inevitably expires as device is only suspended, left in some bag/drawer for a while, etc.  date comes back upon next boot as 01/01/1970 and isnt corrected)

How would you correct that? how to use some minimalistic TUI-interface based script that would allow to set time with the d-pad, or smth?

**Comment by gabuzom** [comment]
After some research heere is what i found:

- timesyncd is running on Amberelec. It is controlled by systemd and enables NTPP sync.
- there is an option in timesyncd to save time \o/
"/var/lib/systemd/timesync/clock
The modification time ("mtime") of this file is updated on each successful NTP synchronization or after each SaveIntervalSec= time interval, as specified in [timesyncd.conf(5)](https://man.archlinux.org/man/timesyncd.conf.5.en). At the minimum, it will be set to the systemd build date. It is used to ensure that the system clock remains roughly monotonic across reboots, in case no local RTC is available."
- unfortunately, /var/lib/systemd/timesync/clock doesnt exist, and /etc/systemd/timesyncd.conf is not writable. 
-  wrote a /etc/systemd/timesyncd.conf.d/save.conf containing 
SaveIntervalSec=600 # (to save time every 10 minutes) 

will see what happens upon reboot and do further tests...

**Comment by gabuzom** [comment]
hmmm all i get from
`systemctl status systemd-timesyncd.service`

is
```
○ systemd-timesyncd.service - Network Time Synchronization
     Loaded: loaded (/usr/lib/systemd/system/systemd-timesyncd.service; disabled; vendor preset: disabled)
    Drop-In: /usr/lib/systemd/system/systemd-timesyncd.service.d
             └─depend-on-kernel-ip-config.conf
     Active: inactive (dead)
  Condition: start condition failed at Wed 2022-12-14 12:12:50 CET; 1s ago
             └─ ConditionPathExists=/dev/.kernel_ipconfig was not met
       Docs: man:systemd-timesyncd.service(8)

Dec 14 11:56:10 R systemd[1]: Network Time Synchronization was skipped because of a failed condition check (ConditionPathExists=/dev/.kernel_ipconfig).
Dec 14 12:09:43 R systemd[1]: Network Time Synchronization was skipped because of a failed condition check (ConditionPathExists=/dev/.kernel_ipconfig).
Dec 14 12:10:18 R systemd[1]: Network Time Synchronization was skipped because of a failed condition check (ConditionPathExists=/dev/.kernel_ipconfig).
Dec 14 12:12:03 R systemd[1]: Network Time Synchronization was skipped because of a failed condition check (ConditionPathExists=/dev/.kernel_ipconfig).
Dec 14 12:12:08 R systemd[1]: Network Time Synchronization was skipped because of a failed condition check (ConditionPathExists=/dev/.kernel_ipconfig).
Dec 14 12:12:41 R systemd[1]: Network Time Synchronization was skipped because of a failed condition check (ConditionPathExists=/dev/.kernel_ipconfig).
Dec 14 12:12:50 R systemd[1]: Network Time Synchronization was skipped because of a failed condition check (ConditionPathExists=/dev/.kernel_ipconfig).

```

any idea? suggestion?

**Reply by medeirosT** [reply]
@gabuzom our "/" partition is made using SquashFS (Read Only).

If you want to make changes to this file, you could:
- Either edit your current squashFS image
- Build the firmware and test the changes! 

Let me know if I can help out! :)

**Comment by dhwz** [comment]
@gabuzom You'll be able to change the time/date in the next coming pre-release image. Just call the newly added tool from tools.

**Reply by gabuzom** [reply]
Amazing!! I got stuck at not having timesyncd running, not finding any cron-like scheme to run a stupid imperfect script i was writing, etc... and gave up :/

thanks!!
